documentation/src/docs/recovery-procedures/table-segment-recovery.md

```diff
@@ -0,0 +1,185 @@
+
+
+# Root Cause
+
+The above issue has been seen in conjuntion with a CacheFullException. That is we see a cacheFullException first and couple of minutes later is when we see the DataCorruptionException described above. Although we have not been able to do a exact root cause analysis on why such a corruption occured we suspect it could be linked to us getting
+a cacheFullException. That is the BTree index pages accessed are also cached in Pravega's internal cache. And trying to access these pages when the cache is full could have led to some wrong data being read, and eventually wrong data being written to the underlying Pravega segments. 
+It could also be an issue pertaining to the underlying tier-2 storage that Pravega uses. That is some kind of corruption having occured in the storage system that Pravega uses that could have caused such an exception.
+However we do not know the exact root cause of what might have led to this and that is the reason this Recovery procudure exists.
+
+
+
+# Repair Procedure
+
+The repair procedure aims at recovering Pravega from such a situation. Here is what the recovery procedure aims to do. Recall that the corruption that we see occurs in the attribute index segment which is just an index of the data that exists in the associated primary table segment( holding key-value pairs ) and that this primary table segment is perfectly fine. So what if we re-read this main table segment and generate an index again which is clean. Thats what the repair procedure does. It revolves around re-reading 
+the key values pairs in the main table segment, feeding them to a in-process Pravega. Doing so would result in an attribute index segment generated into the storage directory that this in-process Pravega points to. 
+
+We replace this generated clean attribute index segment back to the Pravega cluster which had the datacoruption, therby freeing it from the issue.
+
+
+# Detailed Steps
+
+1) From the error, first determine the segment name:-
+```
+Caused by: io.pravega.common.util.IllegalDataFormatException: [AttributeIndex[3-423]] Wrong footer information. RootPage Offset (3842188288) + Length (142858) exceeds Footer Offset (15007837).
+```
+Here AttributeIndex[3-423] indicates container 3 and segment id 423.
+
+Find out from the logs what is the name of the segment with id 423. To do so one can grep "MapstreamSegmentId" and pick the one for the segment id in question.
+
+2) Change to the tier-2 directory having the segment chunks. Usually the root of this directory is /mnt/tier-2. And the table segment chunks can be found under `/mnt/tier2/_system/_tables`.
```

Comment:
> `Change to the tier-2 directory having the segment chunks`

Do you mean "go" or `cd`?

Also, all this explanation relies on using FileSystem as storage. It would be important to note that at the beginning of the explanation and to one test for the same recovery process but using Object Storage as LTS.

Reply:
have stated specifically before starting the detailed steps now about the assumption made on filesystem as tier-2.